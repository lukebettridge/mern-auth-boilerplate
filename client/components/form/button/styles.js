import styled, { css } from "styled-components";

const Button = styled.button`
	background-color: #f8f8f8;
	border: 1px solid #eeeeee;
	cursor: pointer;
	font-size: 13pt;
	font-weight: 600;
	max-width: 350px;
	padding: 8px 20px;
	width: 100%;

	${props =>
		props.as === "a" &&
		css`
			background-color: transparent;
			color: #596f86;
			display: block;
			font-weight: 400;
			max-width: none;
			text-decoration: none;
			&:hover {
				text-decoration: underline;
			}

			${props.secondary &&
				css`
					color: #aaaaaa;
					font-size: 11pt;
					font-weight: 300;
				`}
		`}

	${props =>
		props.mt === "s" &&
		css`
			margin-top: 16px;
		`}

	${props =>
		props.mb === "s" &&
		css`
			margin-bottom: 16px;
		`}
`;

module.exports = { Button };
