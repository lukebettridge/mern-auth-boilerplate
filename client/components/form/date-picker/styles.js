import styled, { css } from "styled-components";

const DatePickerContainer = styled.div`
	${props =>
		props.inError &&
		css`
			& + label {
				color: #db1802;
			}
		`}

	${props =>
		(props.value || props.focused) &&
		css`
			& + label {
				font-size: 95%;
				transform: translate3d(0, -100%, 0);
				opacity: 1;
			}
		`}
`;

const Error = styled.p`
	color: #db1802;
	font-size: 10pt;
	font-weight: 300;
	margin-bottom: 16px;
	margin-top: 0;
	&:first-letter {
		text-transform: capitalize;
	}
`;

const Container = styled.div`
	position: relative;

	${props =>
		props.label &&
		css`
			margin-top: 30px;
		`}

	${props =>
		props.mb &&
		css`
			margin-bottom: 30px;
		`}

	.SingleDatePicker,
	.SingleDatePickerInput,
	.DateInput {
		display: block;
		width: 100%;
	}

	.DateInput_input {
		background-color: transparent;
		border-bottom: 1px solid #e9ecef;
		font-size: 11pt;
		font-weight: 300;
		letter-spacing: 1px;
		margin-bottom: 8px;
		padding: 12px 10px 5px;
		transition: border-color 100ms;
		width: 100%;

		&:hover {
			border-color: #b3b3b3;
		}

		${props =>
			props.inError &&
			css`
				border-color: #db1802;
			`}
	}
`;

const Label = styled.label`
	color: #999999;
	left: 0;
	opacity: 0.5;
	font-weight: 300;
	padding-left: 10px;
	padding-top: 10px;
	position: absolute;
	top: 0;
	transition: all 200ms;
`;

export { Container, DatePickerContainer, Error, Label };
