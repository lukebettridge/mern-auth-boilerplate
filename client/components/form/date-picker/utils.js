const validate = ({ friendlyName, isRequired, name, value }) => {
	let error = "";
	let prefix = friendlyName || name || "This";
	prefix = prefix.charAt(0).toUpperCase() + prefix.slice(1);

	if (isRequired && (!value || value.replace(/^\s+|\s+$/g, "").length === 0)) {
		error = `${prefix} field is required`;
	}
	return error;
};

export { validate };
