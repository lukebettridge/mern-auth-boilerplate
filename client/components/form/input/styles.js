import styled, { css } from "styled-components";
import breakpoints from "components/styles/breakpoints";

const Error = styled.p`
	color: #db1802;
	font-size: 10pt;
	font-weight: 300;
	margin-bottom: 16px;
	margin-top: 0;
	&:first-letter {
		text-transform: capitalize;
	}
`;

const Input = styled.input`
	background-color: transparent;
	border: 0;
	border-bottom: 1px solid #e9ecef;
	box-shadow: none;
	font-size: 11pt;
	font-weight: 300;
	margin-bottom: 8px;
	outline: none;
	padding: 12px 10px;
	width: 100%;

	&:focus + label {
		font-size: 75%;
		transform: translate3d(0, -100%, 0);
		opacity: 1;
	}

	${props =>
		props.value &&
		css`
			& + label {
				font-size: 75%;
				transform: translate3d(0, -100%, 0);
				opacity: 1;
			}
		`}

	${props =>
		props.inError &&
		css`
			border-color: #db1802;

			& + label {
				color: #db1802;
			}
		`}
`;

const InputContainer = styled.div`
	position: relative;
	width: 100%;

	@media (min-width: ${breakpoints.m}) {
		min-width: 250px;
	}

	${props =>
		props.label &&
		css`
			margin-top: 20px;
		`}

	${props =>
		props.mb &&
		css`
			margin-bottom: 30px;
		`}
`;

const Label = styled.label`
	left: 0;
	opacity: 0.5;
	font-weight: 300;
	padding-left: 10px;
	padding-top: 10px;
	position: absolute;
	top: 0;
	transition: all 200ms;
`;

export { Error, Input, InputContainer, Label };
