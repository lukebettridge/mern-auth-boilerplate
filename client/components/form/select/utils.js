const validate = ({ friendlyName, isMulti, isRequired, name, value }) => {
	let error = "";
	const prefix = friendlyName || name || "This";

	if (typeof value === "object") value = value.value;

	if (
		isRequired &&
		(!value ||
			(isMulti && value.length === 0) ||
			(!isMulti && value.replace(/^\s+|\s+$/g, "").length === 0))
	) {
		error = `${prefix} field is required`;
	}
	return error;
};

export { validate };
