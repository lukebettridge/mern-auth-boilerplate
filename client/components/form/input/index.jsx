import React, {
	forwardRef,
	useEffect,
	useImperativeHandle,
	useState
} from "react";
import PropTypes from "prop-types";

import * as utils from "./utils";
import * as S from "./styles";

const Input = forwardRef((props, ref) => {
	const [state, setState] = useState({
		error: props.error || "",
		value: props.value || ""
	});

	useEffect(() => {
		const { error } = props;
		setState(prev => ({ ...prev, error }));
	}, [props.error]);

	useImperativeHandle(ref, () => ({ validate, value: state.value }));

	const onBlur = e => {
		validate(true);
		if (props.onBlur) props.onBlur(e);
	};

	const onChange = e => {
		const value = e.target ? e.target.value : state.value;
		setState(prev => ({ ...prev, error: "", value }));
		if (props.onChange) props.onChange(e);
	};

	const validate = (onBlur = false) => {
		const error = utils.validate(props);
		setState(prev => ({
			...prev,
			error: !onBlur ? props.error || error : error
		}));
		return error;
	};

	const id = props.id || props.name;

	return (
		<S.InputContainer className="inputContainer" {...props}>
			<S.Input
				{...props}
				id={id}
				inError={!!state.error}
				onBlur={onBlur}
				onChange={onChange}
				ref={ref}
				value={state.value}
			/>
			{props.label && <S.Label htmlFor={id}>{props.label}</S.Label>}
			{!!state.error && <S.Error>{state.error}</S.Error>}
		</S.InputContainer>
	);
});

Input.propTypes = {
	error: PropTypes.string,
	friendlyName: PropTypes.string,
	id: PropTypes.string,
	isRequired: PropTypes.bool,
	label: PropTypes.string,
	max: PropTypes.number,
	min: PropTypes.number,
	name: PropTypes.string,
	onBlur: PropTypes.func,
	onChange: PropTypes.func,
	pattern: PropTypes.any,
	placeholder: PropTypes.string,
	type: PropTypes.string,
	value: PropTypes.any
};

Input.defaultProps = {
	placeholder: "",
	type: "text"
};

export default Input;
