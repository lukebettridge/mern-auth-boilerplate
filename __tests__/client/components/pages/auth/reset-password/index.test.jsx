import React from "react";
import { act } from "react-dom/test-utils";
import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import { MemoryRouter } from "react-router-dom";

import { Error } from "components/form/input/styles";
import * as utils from "components/pages/auth/reset-password/utils";

import ResetPassword from "components/pages/auth/reset-password";

let wrapper;
let subject;
const formEvent = { preventDefault: jest.fn() };
jest.mock("components/pages/auth/reset-password/utils", () => ({
	resetPassword: jest.fn()
}));

describe("ResetPassword component", () => {
	beforeEach(() => {
		jest.resetAllMocks();
	});

	afterEach(() => {
		if (wrapper) wrapper.unmount();
	});

	it("snapshot renders", async () => {
		mountWrapper();
		expect(toJSON(subject)).toMatchSnapshot();
	});

	it("prevents submission of invalid form", () => {
		mountWrapper();

		subject.find("form").simulate("submit", formEvent);

		expect(formEvent.preventDefault).toHaveBeenCalled();
		expect(utils.resetPassword).not.toHaveBeenCalled();
	});

	it("allows submission of valid form", () => {
		mountWrapper();
		populateForm();

		subject.find("form").simulate("submit", formEvent);

		expect(formEvent.preventDefault).toHaveBeenCalled();
		expect(utils.resetPassword).toHaveBeenCalledWith(
			{
				resetKey: "foo",
				newPassword: "password",
				newPassword2: "password"
			},
			expect.any(Function),
			expect.any(Function)
		);
	});

	it("catch error with response", () => {
		mountWrapper();
		populateForm();

		subject.find("form").simulate("submit", formEvent);
		act(() => {
			utils.resetPassword.mock.calls[0][2]({
				response: { data: { resetKey: "foo" } }
			});
		});
		updateSubject();

		expect(formEvent.preventDefault).toHaveBeenCalled();

		expect(subject.find(Error).length).toEqual(1);
		expect(subject.find(Error).text()).toEqual("foo");
	});

	it("catch error without response", () => {
		mountWrapper();
		populateForm();

		subject.find("form").simulate("submit", formEvent);
		act(() => {
			utils.resetPassword.mock.calls[0][2]({ response: null });
		});
		updateSubject();

		expect(formEvent.preventDefault).toHaveBeenCalled();

		expect(subject.find(Error).length).toEqual(0);
	});
});

const mountWrapper = () => {
	wrapper = mount(
		<MemoryRouter>
			<ResetPassword {...baseProps} />
		</MemoryRouter>
	);
	subject = wrapper.find(ResetPassword);
};

const updateSubject = () => {
	wrapper.update();
	subject = wrapper.find(ResetPassword);
};

const populateForm = (newPassword = "password", newPassword2 = "password") => {
	subject
		.find("input")
		.at(0)
		.simulate("change", {
			target: { name: "newPassword", value: newPassword }
		});
	subject
		.find("input")
		.at(1)
		.simulate("change", {
			target: { name: "newPassword2", value: newPassword2 }
		});
	updateSubject();
};

const baseProps = {
	match: {
		params: {
			resetKey: "foo"
		}
	}
};
