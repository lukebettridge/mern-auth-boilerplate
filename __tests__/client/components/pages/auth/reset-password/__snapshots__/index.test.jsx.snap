// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword component snapshot renders 1`] = `
.c5 {
  background-color: transparent;
  border: 0;
  border-bottom: 1px solid #e9ecef;
  border-radius: 0;
  box-shadow: none;
  font-size: 11pt;
  font-weight: 300;
  margin-bottom: 8px;
  outline: none;
  padding: 12px 10px;
  -webkit-transition: border-color 100ms;
  transition: border-color 100ms;
  width: 100%;
}

.c5:hover {
  border-color: #b3b3b3;
}

.c5:focus {
  border-color: #2684ff;
}

.c5:focus + label {
  font-size: 75%;
  -webkit-transform: translate3d(0,-100%,0);
  -ms-transform: translate3d(0,-100%,0);
  transform: translate3d(0,-100%,0);
  opacity: 1;
}

.c4 {
  position: relative;
  width: 100%;
  margin-top: 30px;
}

.c7 {
  position: relative;
  width: 100%;
  margin-top: 30px;
  margin-bottom: 30px;
}

.c6 {
  left: 0;
  opacity: 0.5;
  font-weight: 300;
  padding-left: 10px;
  padding-top: 10px;
  position: absolute;
  top: 0;
  -webkit-transition: all 200ms;
  transition: all 200ms;
}

.c2 {
  background-color: #ffffff;
  border: 1px solid #e9ecef;
  margin-bottom: 20px;
  padding: 40px 30px;
  text-align: center;
}

.c0 {
  margin: 0 auto;
  min-height: 100vh;
  padding: 32px 15px;
  width: 100vw;
  max-width: 540px;
}

.c1 {
  color: #212529;
  font-size: 30pt;
  font-weight: 900;
  -webkit-letter-spacing: -1px;
  -moz-letter-spacing: -1px;
  -ms-letter-spacing: -1px;
  letter-spacing: -1px;
  margin-top: 0;
  margin-bottom: 32px;
}

.c3 {
  color: #666666;
  font-weight: 300;
  line-height: 1.9;
  margin-top: 0;
  margin-bottom: 16px;
}

.c3 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c10 {
  color: #666666;
  font-weight: 300;
  line-height: 1.9;
  margin-top: 0;
  margin-bottom: 16px;
  color: #cccccc;
  font-size: 10pt;
  line-height: 1.4;
  text-align: center;
}

.c10 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9 {
  color: #489be8;
  display: block;
  font-size: 11pt;
  font-weight: 300;
  margin-bottom: 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  cursor: pointer;
}

.c9:last-child {
  margin-bottom: 0;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #1188e6;
  border: 1px solid #1288e5;
  border-radius: 2px;
  color: #ffffff;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 10pt;
  font-weight: 300;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 11px;
  outline-width: 3px;
  padding: 10px 17px;
  -webkit-transition: background-color 0.3s;
  transition: background-color 0.3s;
  width: 100%;
}

.c8:disabled {
  cursor: unset;
  opacity: 0.4;
}

@media (min-width:768px) {
  .c4 {
    min-width: 250px;
  }
}

@media (min-width:768px) {
  .c7 {
    min-width: 250px;
  }
}

@media (min-width:768px) {
  .c2 {
    padding: 54px 80px;
    padding-bottom: 50px;
  }
}

@media (min-width:768px) {
  .c0 {
    max-width: calc(100vw - 200px);
    padding: 32px 48px;
  }
}

@media (min-width:768px) {
  .c0 {
    max-width: 700px;
  }
}

<ResetPassword
  match={
    Object {
      "params": Object {
        "resetKey": "foo",
      },
    }
  }
>
  <Layout>
    <styles__Container
      small={true}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "styles__Container-sc-1upuse9-1",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                "margin:0 auto;min-height:100vh;padding:32px 15px;width:100vw;@media (min-width:",
                "768px",
                "){max-width:calc(100vw - 200px);padding:32px 48px;}",
                [Function],
                " ",
                [Function],
                " ",
                [Function],
              ],
            },
            "displayName": "styles__Container",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "styles__Container-sc-1upuse9-1",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        small={true}
      >
        <div
          className="c0"
        >
          <styles__Heading>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styles__Heading-sc-1upuse9-6",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "color:#212529;font-size:30pt;font-weight:900;letter-spacing:-1px;margin-top:0;margin-bottom:32px;",
                    ],
                  },
                  "displayName": "styles__Heading",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styles__Heading-sc-1upuse9-6",
                  "target": "h1",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <h1
                className="c1"
              >
                Webapp
              </h1>
            </StyledComponent>
          </styles__Heading>
          <styles__Box>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styles__Box-sc-1upuse9-0",
                    "isStatic": false,
                    "lastClassName": "c2",
                    "rules": Array [
                      "background-color:#ffffff;border:1px solid #e9ecef;margin-bottom:20px;padding:40px 30px;text-align:center;@media (min-width:",
                      "768px",
                      "){padding:54px 80px;padding-bottom:50px;}",
                      [Function],
                    ],
                  },
                  "displayName": "styles__Box",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styles__Box-sc-1upuse9-0",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="c2"
              >
                <styles__Paragraph
                  mb="l"
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "styles__Paragraph-sc-1upuse9-8",
                          "isStatic": false,
                          "lastClassName": "c10",
                          "rules": Array [
                            "color:#666666;font-weight:300;line-height:1.9;margin-top:0;margin-bottom:16px;",
                            [Function],
                            " ",
                            [Function],
                            " a{color:inherit;text-decoration:none;&:hover{text-decoration underline;}}",
                          ],
                        },
                        "displayName": "styles__Paragraph",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "styles__Paragraph-sc-1upuse9-8",
                        "target": "p",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    mb="l"
                  >
                    <p
                      className="c3"
                    >
                      Fill out the form below and click submit to reset your password.
                    </p>
                  </StyledComponent>
                </styles__Paragraph>
                <form
                  noValidate={true}
                  onSubmit={[Function]}
                >
                  <ForwardRef
                    isRequired={true}
                    label="New Password"
                    name="newPassword"
                    onChange={[Function]}
                    placeholder=""
                    type="password"
                    value=""
                  >
                    <styles__InputContainer
                      className="inputContainer"
                      isRequired={true}
                      label="New Password"
                      name="newPassword"
                      onChange={[Function]}
                      placeholder=""
                      type="password"
                      value=""
                    >
                      <StyledComponent
                        className="inputContainer"
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "styles__InputContainer-sc-223pk0-2",
                              "isStatic": false,
                              "lastClassName": "c7",
                              "rules": Array [
                                "position:relative;width:100%;@media (min-width:",
                                "768px",
                                "){min-width:250px;}",
                                [Function],
                                " ",
                                [Function],
                              ],
                            },
                            "displayName": "styles__InputContainer",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "styles__InputContainer-sc-223pk0-2",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        isRequired={true}
                        label="New Password"
                        name="newPassword"
                        onChange={[Function]}
                        placeholder=""
                        type="password"
                        value=""
                      >
                        <div
                          className="c4 inputContainer"
                          label="New Password"
                          name="newPassword"
                          onChange={[Function]}
                          placeholder=""
                          type="password"
                          value=""
                        >
                          <styles__Input
                            id="newPassword"
                            inError={false}
                            isRequired={true}
                            label="New Password"
                            name="newPassword"
                            onBlur={[Function]}
                            onChange={[Function]}
                            placeholder=""
                            type="password"
                            value=""
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "styles__Input-sc-223pk0-1",
                                    "isStatic": false,
                                    "lastClassName": "c5",
                                    "rules": Array [
                                      "background-color:transparent;border:0;border-bottom:1px solid #e9ecef;border-radius:0;box-shadow:none;font-size:11pt;font-weight:300;margin-bottom:8px;outline:none;padding:12px 10px;transition:border-color 100ms;width:100%;&:hover{border-color:#b3b3b3;}&:focus{border-color:#2684ff;}&:focus + label{font-size:75%;transform:translate3d(0,-100%,0);opacity:1;}",
                                      [Function],
                                      " ",
                                      [Function],
                                    ],
                                  },
                                  "displayName": "styles__Input",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "styles__Input-sc-223pk0-1",
                                  "target": "input",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={
                                Object {
                                  "current": Object {
                                    "validate": [Function],
                                    "value": "",
                                  },
                                }
                              }
                              id="newPassword"
                              inError={false}
                              isRequired={true}
                              label="New Password"
                              name="newPassword"
                              onBlur={[Function]}
                              onChange={[Function]}
                              placeholder=""
                              type="password"
                              value=""
                            >
                              <input
                                className="c5"
                                id="newPassword"
                                label="New Password"
                                name="newPassword"
                                onBlur={[Function]}
                                onChange={[Function]}
                                placeholder=""
                                type="password"
                                value=""
                              />
                            </StyledComponent>
                          </styles__Input>
                          <styles__Label
                            htmlFor="newPassword"
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "styles__Label-sc-223pk0-3",
                                    "isStatic": false,
                                    "lastClassName": "c6",
                                    "rules": Array [
                                      "left:0;opacity:0.5;font-weight:300;padding-left:10px;padding-top:10px;position:absolute;top:0;transition:all 200ms;",
                                    ],
                                  },
                                  "displayName": "styles__Label",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "styles__Label-sc-223pk0-3",
                                  "target": "label",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                              htmlFor="newPassword"
                            >
                              <label
                                className="c6"
                                htmlFor="newPassword"
                              >
                                New Password
                              </label>
                            </StyledComponent>
                          </styles__Label>
                        </div>
                      </StyledComponent>
                    </styles__InputContainer>
                  </ForwardRef>
                  <ForwardRef
                    friendlyName="Confirm new password"
                    isRequired={true}
                    label="Confirm New Password"
                    mb="m"
                    name="newPassword2"
                    onChange={[Function]}
                    placeholder=""
                    type="password"
                    value=""
                  >
                    <styles__InputContainer
                      className="inputContainer"
                      friendlyName="Confirm new password"
                      isRequired={true}
                      label="Confirm New Password"
                      mb="m"
                      name="newPassword2"
                      onChange={[Function]}
                      placeholder=""
                      type="password"
                      value=""
                    >
                      <StyledComponent
                        className="inputContainer"
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "styles__InputContainer-sc-223pk0-2",
                              "isStatic": false,
                              "lastClassName": "c7",
                              "rules": Array [
                                "position:relative;width:100%;@media (min-width:",
                                "768px",
                                "){min-width:250px;}",
                                [Function],
                                " ",
                                [Function],
                              ],
                            },
                            "displayName": "styles__InputContainer",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "styles__InputContainer-sc-223pk0-2",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        friendlyName="Confirm new password"
                        isRequired={true}
                        label="Confirm New Password"
                        mb="m"
                        name="newPassword2"
                        onChange={[Function]}
                        placeholder=""
                        type="password"
                        value=""
                      >
                        <div
                          className="c7 inputContainer"
                          label="Confirm New Password"
                          name="newPassword2"
                          onChange={[Function]}
                          placeholder=""
                          type="password"
                          value=""
                        >
                          <styles__Input
                            friendlyName="Confirm new password"
                            id="newPassword2"
                            inError={false}
                            isRequired={true}
                            label="Confirm New Password"
                            mb="m"
                            name="newPassword2"
                            onBlur={[Function]}
                            onChange={[Function]}
                            placeholder=""
                            type="password"
                            value=""
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "styles__Input-sc-223pk0-1",
                                    "isStatic": false,
                                    "lastClassName": "c5",
                                    "rules": Array [
                                      "background-color:transparent;border:0;border-bottom:1px solid #e9ecef;border-radius:0;box-shadow:none;font-size:11pt;font-weight:300;margin-bottom:8px;outline:none;padding:12px 10px;transition:border-color 100ms;width:100%;&:hover{border-color:#b3b3b3;}&:focus{border-color:#2684ff;}&:focus + label{font-size:75%;transform:translate3d(0,-100%,0);opacity:1;}",
                                      [Function],
                                      " ",
                                      [Function],
                                    ],
                                  },
                                  "displayName": "styles__Input",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "styles__Input-sc-223pk0-1",
                                  "target": "input",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={
                                Object {
                                  "current": Object {
                                    "validate": [Function],
                                    "value": "",
                                  },
                                }
                              }
                              friendlyName="Confirm new password"
                              id="newPassword2"
                              inError={false}
                              isRequired={true}
                              label="Confirm New Password"
                              mb="m"
                              name="newPassword2"
                              onBlur={[Function]}
                              onChange={[Function]}
                              placeholder=""
                              type="password"
                              value=""
                            >
                              <input
                                className="c5"
                                id="newPassword2"
                                label="Confirm New Password"
                                name="newPassword2"
                                onBlur={[Function]}
                                onChange={[Function]}
                                placeholder=""
                                type="password"
                                value=""
                              />
                            </StyledComponent>
                          </styles__Input>
                          <styles__Label
                            htmlFor="newPassword2"
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "styles__Label-sc-223pk0-3",
                                    "isStatic": false,
                                    "lastClassName": "c6",
                                    "rules": Array [
                                      "left:0;opacity:0.5;font-weight:300;padding-left:10px;padding-top:10px;position:absolute;top:0;transition:all 200ms;",
                                    ],
                                  },
                                  "displayName": "styles__Label",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "styles__Label-sc-223pk0-3",
                                  "target": "label",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                              htmlFor="newPassword2"
                            >
                              <label
                                className="c6"
                                htmlFor="newPassword2"
                              >
                                Confirm New Password
                              </label>
                            </StyledComponent>
                          </styles__Label>
                        </div>
                      </StyledComponent>
                    </styles__InputContainer>
                  </ForwardRef>
                  <Button
                    type="submit"
                  >
                    <styles__Button
                      type="submit"
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "styles__Button-sc-1mykk3k-0",
                              "isStatic": false,
                              "lastClassName": "c8",
                              "rules": Array [
                                "align-items:center;background-color:#1188e6;border:1px solid #1288e5;border-radius:2px;color:#ffffff;cursor:pointer;display:flex;font-size:10pt;font-weight:300;justify-content:center;margin-bottom:11px;outline-width:3px;padding:10px 17px;transition:background-color 0.3s;&:disabled{cursor:unset;opacity:0.4;}",
                                [Function],
                                " ",
                                [Function],
                              ],
                            },
                            "displayName": "styles__Button",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "styles__Button-sc-1mykk3k-0",
                            "target": "button",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        type="submit"
                      >
                        <button
                          className="c8"
                          type="submit"
                        >
                          Submit
                        </button>
                      </StyledComponent>
                    </styles__Button>
                  </Button>
                </form>
                <styles__RouterLink
                  to="/auth/login"
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "styles__RouterLink-sc-1upuse9-9",
                          "isStatic": false,
                          "lastClassName": "c9",
                          "rules": Array [
                            [Function],
                          ],
                        },
                        "displayName": "styles__RouterLink",
                        "foldedComponentIds": Array [],
                        "propTypes": Object {
                          "innerRef": [Function],
                          "onClick": [Function],
                          "replace": [Function],
                          "target": [Function],
                          "to": [Function],
                        },
                        "render": [Function],
                        "styledComponentId": "styles__RouterLink-sc-1upuse9-9",
                        "target": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "displayName": "Link",
                          "propTypes": Object {
                            "innerRef": [Function],
                            "onClick": [Function],
                            "replace": [Function],
                            "target": [Function],
                            "to": [Function],
                          },
                          "render": [Function],
                        },
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    to="/auth/login"
                  >
                    <Link
                      className="c9"
                      to="/auth/login"
                    >
                      <LinkAnchor
                        className="c9"
                        href="/auth/login"
                        navigate={[Function]}
                      >
                        <a
                          className="c9"
                          href="/auth/login"
                          onClick={[Function]}
                        >
                          Go back to login
                        </a>
                      </LinkAnchor>
                    </Link>
                  </StyledComponent>
                </styles__RouterLink>
              </div>
            </StyledComponent>
          </styles__Box>
          <styles__Paragraph
            center={true}
            light={true}
          >
            <StyledComponent
              center={true}
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styles__Paragraph-sc-1upuse9-8",
                    "isStatic": false,
                    "lastClassName": "c10",
                    "rules": Array [
                      "color:#666666;font-weight:300;line-height:1.9;margin-top:0;margin-bottom:16px;",
                      [Function],
                      " ",
                      [Function],
                      " a{color:inherit;text-decoration:none;&:hover{text-decoration underline;}}",
                    ],
                  },
                  "displayName": "styles__Paragraph",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styles__Paragraph-sc-1upuse9-8",
                  "target": "p",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              light={true}
            >
              <p
                className="c10"
              >
                Developed by
                 
                <a
                  href="https://paddl.co.uk/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Paddl
                </a>
                <br />
                . A limited company registered in England and Wales. Registered number: 11992253.
              </p>
            </StyledComponent>
          </styles__Paragraph>
        </div>
      </StyledComponent>
    </styles__Container>
  </Layout>
</ResetPassword>
`;
